monai/apps/reconstruction/networks/nets/utils.py

```diff
@@ -0,0 +1,180 @@
+
+
+def reshape_channel_complex_to_last_dim(x: Tensor) -> Tensor:  # type: ignore
+    """
+    Swaps the complex dimension with the channel dimension so that the network output has 2 as its last dimension
+
+    Args:
+        x: input of shape (B,C*2,H,W) for 2D data or (B,C*2,H,W,D) for 3D data
+
+    Returns:
+        output of shape (B,C,H,W,2) for 2D data or (B,C,H,W,D,2) for 3D data
+    """
+    if x.shape[1] % 2 != 0:
+        raise ValueError(f"channel dimension should be even but ({x.shape[1]}) is odd.")
+
+    if len(x.shape) == 4:  # this is 2D
+        b, c2, h, w = x.shape  # c2 means c*2
+        c = c2 // 2
+        return x.view(b, 2, c, h, w).permute(0, 2, 3, 4, 1)
+
+    elif len(x.shape) == 5:  # this is 3D
+        b, c2, h, w, d = x.shape  # c2 means c*2
+        c = c2 // 2
+        return x.view(b, 2, c, h, w, d).permute(0, 2, 3, 4, 5, 1)
+
+
+def complex_normalize(x: Tensor) -> Sequence:  # type: ignore
+    """
+    Performs layer mean-std normalization for complex data. Normalization is done for each batch member
+    along each part (part refers to real and imaginary parts), separately.
+
+    Args:
+        x: input of shape (B,C,H,W) for 2D data or (B,C,H,W,D) for 3D data
+
+    Returns:
+        A tuple containing
+            (1) normalized output of shape (B,C,H,W) for 2D data or (B,C,H,W,D) for 3D data
+            (2) mean
+            (3) std
+    """
+    if len(x.shape) == 4:  # this is 2D
+        b, c, h, w = x.shape
+        x = x.contiguous().view(b, 2, c // 2 * h * w)
+        mean = x.mean(dim=2).view(b, 2, 1, 1, 1).expand(b, 2, c // 2, 1, 1).contiguous().view(b, c, 1, 1)
+        std = x.std(dim=2, unbiased=False).view(b, 2, 1, 1, 1).expand(b, 2, c // 2, 1, 1).contiguous().view(b, c, 1, 1)
+        x = x.view(b, c, h, w)
+        return (x - mean) / std, mean, std
+
+    elif len(x.shape) == 5:  # this is 3D
+        b, c, h, w, d = x.shape
+        x = x.contiguous().view(b, 2, c // 2 * h * w * d)
+        mean = x.mean(dim=2).view(b, 2, 1, 1, 1, 1).expand(b, 2, c // 2, 1, 1, 1).contiguous().view(b, c, 1, 1, 1)
+        std = (
+            x.std(dim=2, unbiased=False)
+            .view(b, 2, 1, 1, 1, 1)
+            .expand(b, 2, c // 2, 1, 1, 1)
+            .contiguous()
+            .view(b, c, 1, 1, 1)
+        )
+        x = x.view(b, c, h, w, d)
+        return (x - mean) / std, mean, std  # type: ignore
+
+
+def inverse_complex_normalize(x: Tensor, mean: float, std: float) -> Tensor:  # type: ignore
+    """
+    Reverses the normalization done by complex_normalize
+
+    Args:
+        x: input of shape (B,C,H,W) for 2D data or (B,C,H,W,D) for 3D data
+        mean: mean before normalization
+        std: std before normalization
+
+    Returns:
+        denormalized output of shape (B,C,H,W) for 2D data or (B,C,H,W,D) for 3D data
+    """
+    return x * std + mean
+
+
+def pad(x: Tensor) -> Sequence:  # type: ignore
```

Comment:
This function needs to be renamed... `pad()` is too general.

Reply:
From the example, it looks to me that this is https://github.com/Project-MONAI/MONAI/blob/2c89142a14831a7f39d8c1acfaf1b83074aa041f/monai/transforms/croppad/array.py#L322

Could you double check it? Thanks!

Reply:
Cool, I thought `DivisiblePad` was doing something different than our pad, but it's the same as you mentioned and I just fixed it.